Added NoEmptyPhpdocFixer

SpacePossum opened the pull request:
closes https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/1688

keradus commented:
`NoEmptyPhpdocFixer` ?

keradus commented on Symfony/CS/Tests/Fixer/Contrib/EmptyPhpdocFixerTest.php:
```diff
@@ -0,0 +1,77 @@
+                     *
+                     */
+
+                    /** ***
```
This is not empty docblock. it's odd, but not empty. The Summary is `***`.

SpacePossum replied:
I know, I don't like those in my projects.. do you want these?

keradus replied:
It's a user case thing. Taking things strictly - this is part of docblock, so it's not empty...
I don't like them too, but still...

GrahamCampbell replied:
Maybe we should define empty more loosly has having nothing useful in them?

SpacePossum replied:
very well, let me try out some more regex stuff.

keradus commented on README.rst:
```diff
@@ -615,6 +615,10 @@ Choose from the list of available fixers:
                         construct to print if
                         possible.
 
+* **empty_phpdoc** [contrib]
```
Symfony?

SpacePossum replied:
I'll ask them if they reply on my PR overthere, till than I'm not going to the trouble to check it all out. If they want it we can always move it to SF level.

keradus replied:
No need to ask at first, I could send the PR (in the and you can treat PR as extra type of issue with code attached).
But seriously, I don't think there will be any violation for that rule.

The only case I met and why I want that fixer is our annotation cleaning fixers - eg when we remove `@access` annotation and there is nothing left in docblock

keradus commented on Symfony/CS/Tests/Fixer/AbstractFixerTestBase.php:
```diff
@@ -36,6 +36,9 @@ public static function setUpBeforeClass()
     {
         $name = 'Symfony\CS\Fixer'.substr(get_called_class(), strlen(__NAMESPACE__), -strlen('Test'));
+        if (!class_exists($name)) {
+            $this->fail(sprintf('Fixer "%s" for test not found.', $name));
```
???

SpacePossum replied:
DX, readability of the fail is better than a fatal stack dump

keradus replied:
fail mark the test as failed, and here we are not in test...

GrahamCampbell replied:
Just throw an exaception?

SpacePossum commented:
closing till I figured out how to fix it (https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/1765#discussion_r52298636)

keradus commented:
@SpacePossum , please play with utests so the utests will define everything that should be fixed and how - when it will be done (or it is already?) I will help with regexp

SpacePossum commented:
Renamed, replaced the `fail` with `throw exception` (thanks @GrahamCampbell ) and added a bunch of integration tests (include case for removal of `@access` thanks @keradus).
Given that some of the tests are still running, I think the one thing (and most important thing) is to figure out the rule on which to remove a PHPDoc.
I like it as is, but I see the point @keradus is making.
It will be tricky to tweak though, so lets list some cases here.
I think we agree on these:
```php
/** */

/** 
*/

/** 
 *
 *
 */
```
However, the fixer also removes:
```php
/**
**/

/*******
 *
 *****/

/**
 * *
 */
```
I'm happy with both sets, so let me know were to aim for :)

keradus commented:
For 6 cases you provided I woudn't touch only the last one, in which `*` is summary part of PHPDoc. Odd, not nice, but still valid.

gharlan commented:
I like it as-is.

keradus commented:
@SpacePossum All cases you provided are also in utests, am I right ?

SpacePossum commented:
@keradus yes I think so, I checked and cannot think of anymore other cases, let me know I you have any in mind.

keradus commented:
Cool!
Please, move it to Symfony level

SpacePossum commented:
>Please, move it to Symfony level

:+1:  done :)

keradus commented:
Thank you @SpacePossum.
